### ISSUE TITLE
Template direcrory structure

### ISSUE THREAD

**Issue by mrLSD** [kaj/ructe#5]
@kaj Thank you for really awesome Template Engine.
I provide several tests and it is extremely fast, compile-time template checking.
But it is inconvenient thing - I can't structuring my templates.
Fro example:
```
templates/
    admin/
        main.rs.html
    layouts/
        _header.rs.html
        main.rs.html
```

Are you planning implement it?

**Comment by kaj** [comment]
Yes, a directory structure should absolutley be allowed!  There might be some API questions that needs answers first, though.  In your example, I think one call to `compile_templates` from `build.rs` should be sufficient.  Should the directories be reflected as Rust modules?  Yes, I think they should.  So your directory structure should create the functions `templates::admin::main`, `templates::layouts::_header` and `templates::layouts::main`.  Right?

I guess the name starting with an underscore should be private, which would mean it can only be used from other templates in the same directory and sub-directories of that (in this case only from `templates::layouts::main`).  But that could always be added later.  Maybe it is even better to just have it public, with the underscore as a hint that it is not a full page, or something.

**Comment by mrLSD** [comment]
Exactly!
Did you mean future API?

**Comment by kaj** [comment]
Yes, API questions as in "how should the API for using this look".

**Comment by kaj** [comment]
Merged to master.  I decided about any special handling of filenames starting with an underscore, at least for now.

Any further comments before I make a relase?

**Comment by mrLSD** [comment]
I looked to commits and exemples. It looks like ready to release.
@kaj, thank you for awesome library!

**Comment by kaj** [comment]
Released as part of version 0.2.0, which is on crates.io now.  Thank you, @mrLSD, for the suggestion.
